lib/ansible/config/base.yml

```diff
@@ -1844,6 +1844,15 @@ TAGS_SKIP:
   ini:
   - {key: skip, section: tags}
   version_added: "2.5"
+TASK_TIMEOUT:
+  name: Task Timeout
+  default: 0
+  description:
+    - Set the maximum time (in seconds) that a task can run for.
+  env: [{name: ANSIBLE_TASK_TIMEOUT}]
```

Comment:
Being able to set this globally for every task feels odd to me, but I might just not be creative enough to think of a case where it would be useful.

Reply:
when dealing with 40k servers you might want it globally, at play/block/role or task level, didn't make much sense to me to restrict it to only settable in one place